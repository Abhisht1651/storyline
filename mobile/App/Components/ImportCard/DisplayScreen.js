import React from 'react'
import { View } from 'react-native'
import ImportCard from './ImportCard'
import styles from '../Styles/ImportCardStyles/DisplayScreenStyle'

const DisplayScreen = () => {
  return (
    <View style={styles.container}>
      <View style={styles.box}>
        <ImportCard />
      </View>
    </View>
  )
}

export default DisplayScreen
