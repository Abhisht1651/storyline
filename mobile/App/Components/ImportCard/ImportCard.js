// @flow

import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import styles from '../Styles/ImportCardStyles/ImportCardStyle'
import PhoneIcon from '../../Images/Icons/Icon.svg'

type Props = {
  onPress: () => {}
}

const ImportCard = (props: Props) => {
  const { onPress } = props
  return (
    <TouchableOpacity style={styles.container} activeOpacity={1} onPress={onPress}>
      <View style={styles.upperBlock}>
        <PhoneIcon height="100%" />
      </View>
      <View style={styles.middleBlock} activeOpacity={1}>
        <Text style={styles.middleBlockText}>Import contacts from your phone</Text>
      </View>
      <View style={styles.lastBlock} activeOpacity={1}>
        <Text style={styles.lastBlockText}>Get access to my adresss book</Text>
      </View>
    </TouchableOpacity>
  )
}

export default ImportCard
