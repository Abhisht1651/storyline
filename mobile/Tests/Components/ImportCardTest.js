import { shallow } from 'enzyme'
import React from 'react'
import ImportCard from '../../App/Components/ImportCard/ImportCard'

describe('ImportCard', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<ImportCard />)
      expect(component).toMatchSnapshot()
    })
  })
})

it('should render the Text Component', () => {
  const wrapper = shallow(<ImportCard />)
  expect(wrapper.find('Text')).toHaveLength(2)
})
