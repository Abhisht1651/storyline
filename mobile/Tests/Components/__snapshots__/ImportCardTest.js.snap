// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportCard Rendering should match to snapshot 1`] = `
<ForwardRef
  activeOpacity={1}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#DCDCE9",
      "borderRadius": 24.642857142857142,
      "borderWidth": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "height": "30%",
        "marginBottom": 13.142857142857142,
        "marginTop": 49.285714285714285,
      }
    }
  >
    <SvgMock
      height="100%"
    />
  </View>
  <View
    activeOpacity={1}
    style={
      Object {
        "alignItems": "center",
        "height": "22%",
        "justifyContent": "center",
        "marginBottom": 6.571428571428571,
        "width": "61.53%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 31,
          "letterSpacing": -0.12,
          "lineHeight": 36.14285714285714,
          "textAlign": "center",
        }
      }
    >
      Import contacts from your phone
    </Text>
  </View>
  <View
    activeOpacity={1}
    style={
      Object {
        "height": "8.5%",
        "width": "61.53%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Light",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      Get access to my adresss book
    </Text>
  </View>
</ForwardRef>
`;
